### PR TITLE
Add documentation and plugin notice about the end of the plugin version

### LINKED ISSUE #2794: Timber 2.0 and the plugin version
With 2.0 around the corner. We would like to see people trying out Timber 2.0 in their projects. But from a response lately:
> What about support for the Timber plugin? Will the plugin be updated with the release of Timber 2.0?
> 
> I have some older sites that are using the plugin and we are not able to switch to installing Timber in our theme at this time.

_Originally posted by @okadots in https://github.com/timber/timber/issues/2793#issuecomment-1718297524_
            
We can see that there is quite some attention for the removal of the plugin version.

### Why we are dropping support for installing Timber 2.0 as a plugin
In #2005 @gchtr lays out why we are dropping the plugin support. The main takeaways:

- Timber was never a "plugin", so easy to update with the press of a button in the admin environment.
- Supporting Timber also as a plugin makes it harder to introduce breaking changes. Since Timber 2.0 introduces a lot of breaking changes, this was an issue as well.

### What we have to do to signal the big upcoming change
As stated by @rmens 

>  Should there be a big fat warning on the plug-in page on Wordpress.org that it’s essentially sunset? You don’t want people starting with it with 2.0 around the corner.

_Originally posted by rmens in https://github.com/timber/timber/issues/2793#issuecomment-1718834444_

It might be a good idea to make to changes to the readme of [Timber on WordPress.org](https://nl.wordpress.org/plugins/timber-library/) to signal people to migrate over to the composer version of Timber instead.

@gchtr and @jarednova , what is your take on this?

### What you, the people using Timber need to make the transition
We would like to know what is holding you back making the switch to the composer version of Timber. We cannot promise to take all issues into consideration but if we spot some clear trends as of why people haven't made the transition yet, we might be able to write some extra documentation, make a video, or something along those lines. 

### TODOs for phasing out the plugin version

- [x] Update the Timber page at [Upstatement](https://upstatement.com/timber/). Primarily update the install link. @jarednova (change is made and should be deployed in the next few days)
- [x] Pinned issue in https://wordpress.org/support/plugin/timber-library/ @jarednova , I think you have admin access to the plugin at wp.org
- [ ]  [PR pending - create documentation for the upgrade path from switching from the plugin version to the composer version.](https://github.com/timber/timber/pull/2800)
- [ ] PR pending - Add a notification in the plugin row with a message about the support ending for the plugin
- [ ] PR pending - Update the readme.txt and add a message there as well so it ends up at the [plugin wp.org page](https://nl.wordpress.org/plugins/timber-library/)
- [ ] PR pending- Update 1.x docs
- [ ] PR pending - Add message in [readme.md](https://github.com/timber/timber/blob/master/README.md) in main branch

### PR DESCRIPTION
**Pull request by Levdbas**
Related:

- #2794

## Work in progress
This is a first version of this PR where I would like to have input from the core team: @gchtr , @nlemoine and @jarednova . Could you have a look at the wording of the notices in the .txt, .md file? Is that what we want to communicate? Please change/complete my suggested texts and notifications where necessary so that we can release this.

**Example of the plugin notice as currently commited**
afbeelding

### TODO's

- [x] Fill the links in the code changes
- [x] test the upgrade guide with a minimal theme and latest WordPress version
- [x] [Release a plugin version with the new code changes.](https://github.com/timber/timber/issues/2820)

## Issue
We are not going to release a plugin version of Timber anymore. We think that a lot of people will be stuck on the plugin version of Timber 1. With a install base of 50.000+ installations we are talking about a lot of people.

## Solution
Let's inform Timber users about the upcoming end of the Timber plugin and help them with extra guides on how to switch to the Composer based (1.x) version.

## Impact
Minimal on the codebase. This PR adds an inline plugin notification as the only code change.

## Usage Changes
See the whole guide :)

## Considerations

- I made a docs/getting-started/switch-to-composer.md guide. Is this the proper location?
- See the plugin notification I currently created. Is this the way we want to notify the plugin users about this change
- Is the switch-to-composer guide good enough?

## Testing
No

### CODE REVIEW

File: README.md
Comment (jrathert) [inline README.md:23]: I guess we should say something about the fact that v1 will no more supported with PHP 8.2 (has this been decided/announced?).

File: docs/getting-started/setup.md
Comment (jrathert) [inline docs/getting-started/setup.md:11]: See above - say something about PHP 8.2

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:38]: This init call and answering the questions is not needed at all. You just need to issue the "require" line to fetch the right files.

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:43]: Maybe add a comment that composer might ask you to allow plugins and it is safe to say "yes"? Would help people with little composer know-how (like me ;-).

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:47]: Maybe add that the following lines might actually _already be_ in your functions.php (i.e., if your theme is based on the Timber starter theme)

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:65]: May I suggest the following changes: 
When you are 100% sure that you have successfully upgraded your theme in the development website to the composer based version of Timber, you can deploy your theme changes to your live website. Make sure that the new `vendor` folder inside your theme gets deployed as well. If you are using a deployment pipeline, you can commit your changes and push them to your live website. If you are not using a version control system, you can use SCP or FTP to upload your changes to your theme. As a last step you should disable and then remove the Timber v1 plugin from your live website. Congrats - you successfully moved to the composer-based install and are prepared to later upgrade to Timber v2.

File: docs/getting-started/setup.md
Comment (jrathert) [inline docs/getting-started/setup.md:21]: Would take these lines out - we should simply not suggest and describe how to use the plugin...

**Comment by Levdbas** [comment]
Thank you for the feedback @jrathert, I'll pick this up very soon.

File: readme.txt
Comment (gchtr) [inline readme.txt:15]:
```suggestion
You will find an extensive list with guides and the reasons why we are not going to release Timber 2.0 as a plugin anymore.
```

File: readme.txt
Comment (gchtr) [inline readme.txt:17]:
```suggestion
### Switching to the Composer based version
```

File: readme.txt
Comment (gchtr) [inline readme.txt:18]:
```suggestion
* Guide: [How do I switch over from the plugin version to the Composer based version of Timber?](https://timber.github.io/docs/getting-started/switch-to-composer/)
```

File: readme.txt
Comment (gchtr) [inline readme.txt:20]:
```suggestion
* GitHub issue: [Roadmap for Timber 2.0](https://github.com/timber/timber/issues/2741)
```

File: docs/getting-started/setup.md
Comment (gchtr) [inline docs/getting-started/setup.md:15]: Add small updates from above.

File: README.md
Comment (gchtr) [reply README.md:23]: Yes, this is important to note and one of the main reasons why people should make the switch, too.

File: docs/getting-started/setup.md
Comment (gchtr) [inline docs/getting-started/setup.md:21]:
```suggestion
~~You can grab the all-things-included plugin at [WordPress.org](http://wordpress.org/plugins/timber-library/) either through the WordPress site or through the Plugins menu in the backend. Then skip ahead to [using the starter theme](#use-the-starter-theme).~~
```

File: lib/Admin.php
Comment (gchtr) [inline lib/Admin.php:9]: Yes, this is exactly what I meant. This catches the attention of people who are responsible for managing the plugins without being too intrusive.

File: lib/Admin.php
Comment (gchtr) [inline lib/Admin.php:25]: I’d make the first link take fewer words so that it’s easier to see that it’s two links.

```suggestion
		echo '<td colspan="3" class="plugin-update colspanchange"><div class="update-message notice-error notice inline notice-warning notice-alt"><p>We recommend <a target="_blank" href="#">switching to the Composer based version</a> of Timber. <a href="#" target="_blank">Why you need to switch.</a></p></div></td>';
```

**Review by gchtr — COMMENTED** [review]
This is fantastic, @Levdbas. Thanks so much, I’m sure this will help so many people to make the switch.

> I made a docs/getting-started/switch-to-composer.md guide. Is this the proper location?

Yes, this looks good to me!

> See the plugin notification I currently created. Is this the way we want to notify the plugin users about this change

Yes, I would have done it like that.

> Is the switch-to-composer guide good enough?

Oh yes it is.

Thanks also to @jrathert for taking the time to read through this as someone who hasn’t a lot of experience with Composer 🙌.

File: README.md
Comment (Levdbas) [reply README.md:23]: Are we calling it Timber 2 or Timber 2.0? Added a link as well.

Comment (Levdbas) [reply README.md:23]: I tried to add it to the notice as well. @jrathert @gchtr. I also clearified that the upgrade guide is only the first step in upgrading to Timber 2.0

File: lib/Admin.php
Comment (Levdbas) [reply lib/Admin.php:25]: Nice, added the link as well @gchtr

File: docs/getting-started/switch-to-composer.md
Comment (Levdbas) [reply docs/getting-started/switch-to-composer.md:43]: Great suggestion. Added that as well!

**Comment by Levdbas** [comment]
Hi @jrathert  and @gchtr, thanks for all the suggestions. I _think_ I added them all and fixed up a final few todos as well. Maybe lets have one more pass at this?

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:18]:
```suggestion
```

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:15]:
```suggestion
3. Disable the Timber plugin
```

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:22]:
```suggestion
### 1. Get a local development version of your website up and running
We highly recommend doing these steps on a local development version of your website. If you don’t have one yet, you could use [Local by Flywheel](https://localbyflywheel.com/) to get one up and running in a few minutes.
```

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:33]:
```suggestion

Please note that your website will at this point in time no more work as there is no Timber available. This will be fixed in the next step.
```

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:54]:
```suggestion
 * As Timber is installed as a Composer dependency, we need this block
```

File: docs/getting-started/switch-to-composer.md
Comment (jrathert) [inline docs/getting-started/switch-to-composer.md:55]:
```suggestion
 * to load dependencies and initialize Timber.
```

**Review by jrathert — CHANGES_REQUESTED** [review]
Hi @Levdbas - I added a few suggestions (some of the last changes were actually erroneous, I think). Thanks!

**Comment by jrathert** [comment]
> Maybe lets have one more pass at this?

@Levdbas , thanks for the hard work (I am not only new to PHP, Twig, Timber and Composer, but also not too "github-savy" yet, but I keep on learning - I tried to add a few smaller changes/fixes as a review, hope it worked... ;-))

One thing that came to my mind when reading #2804: We always say we drop support for the plugin version of Timber (v1, needless to say). But will we still support the Composer-version of Timber v1? E.g., are there any plans/decision re: updating the code of v1 to make it PHP 8.2-friendlier at least (see #2793)? 

Because I fear if we only tell people now to switch to the Composer version, they might feel safe re: PHP 8.2 as well, and maybe miss that they _need_ to upgrade/change something one more time.

File: docs/getting-started/switch-to-composer.md
Comment (Levdbas) [reply docs/getting-started/switch-to-composer.md:33]: Good addition, I changed it a bit to:

`Please note that you website will throw erros at this point, as there is no Timber available. This will be fixed in the next two steps.`

File: docs/getting-started/switch-to-composer.md
Comment (Levdbas) [reply docs/getting-started/switch-to-composer.md]: I have redone this codeblock to be more in line with the function.php of the [Timber starter theme v2 functions file](https://github.com/timber/starter-theme/blob/2.x/functions.php)

**Review by nlemoine — DISMISSED** [review]
Great work @Levdbas !

**Comment by jarednova** [comment]
This is looking good @Levdbas! Can you tag me when this is merged and ready for deployment to wp.org? Also happy to invite your wp.org user name if you're comfortable inside their SVN environment

**Comment by Levdbas** [comment]
I think we are good to go here! Although I have my own plugin in the .org repo, to say that I am comfortable with the SVN stuff, not really... 👯 

The path forward would be:

1. Merge PR and maybe directly https://github.com/timber/timber/pull/2805 as well 
2. Deploy new version of the docs
3. Check if the links work as intended.
4. Release new plugin version, while at it, maybe update the "tested up to" version as well?
5. The first two steps of this issue: https://github.com/timber/timber/issues/2794 @jarednova

**Review by jarednova — DISMISSED** [review]
This looks great! I just adjusted the emphasis on some of the language at top (emphasizing Composer as the supported method) and tightening the formatting a bit

**Comment by jarednova** [comment]
@Levdbas I feel like you have the best view into sequencing of how we merge this (#2800), #2805 and #2794. Could you proceed with merging this into master when you have the necessary input and dependencies?

Once merged, please create a 1.x release checklist ...

https://github.com/timber/timber/issues/new?assignees=&labels=&projects=&template=release_checklist_1x.md

That will be my cue that things are ready for SVN ~hell~ delight!

**Comment by gchtr** [comment]
> And regarding the support of the Composer Timber v1 version. I am pretty sure we are not going to actively update it anymore. So to make it compatible with PHP 8.2 would be out of scope as well.
> 
> Going though the documentation of this PR and already existing documentation, how would you suggest to communicate that as well? @gchtr , can you chime in on this as well?

I’d say at the moment, we won’t maintain the 1.x version anymore than really necessary.



> The path forward would be:
> 
> 1. Merge PR and maybe directly [docs: updated plugin support part #2805](https://github.com/timber/timber/pull/2805) as well
> 2. Deploy new version of the docs

I have an addition to point 2 in this list: We need to merge this into 2.x as well before we can deploy the docs. Because the docs are built against the 2.x branch.

@Levdbas I’ll let you merge this when you feel that this is ready. From my side, this looks all good. After merging this PR, I can see that we can merge it into 2.x as well and publish a new version of the docs.

**Review by gchtr — APPROVED** [review]

**Comment by Levdbas** [comment]
Alright @gchtr , Its merged! So you will merge the changes made to the MD files into the 2.x branch?

**Comment by aevansgit** [comment]
I'm using Timber v1.23 on a site that we're planning to replace with a new build but not until June or July 2024. Do you think I'll be able to get away with leaving as is?

**Comment by Levdbas** [comment]
Hi @aevansgit , are you using the plugin version? If so, you could try to migrate to the composer based version of 1.23. This way you will be able to receive one final update for the 1.x branch which will improve compatibility with PHP 8.1.

See https://timber.github.io/docs/v1/getting-started/switch-to-composer/

**Pull request merged**
